packages/utils/src/DataUtils.ts

```diff
@@ -8,66 +8,57 @@
- * @param name Contextual name of the address, e.g. "claim owner".
- * @returns Boolean whether the given address string checks out against the Format.
+ * @param input Address string to validate for correct Format.
  */
-export function validateAddress(address: KiltAddress, name: string): boolean {
-  if (typeof address !== 'string') {
+export function verifyKiltAddress(input: unknown): void {
+  if (typeof input !== 'string') {
     throw new SDKErrors.AddressTypeError()
   }
-  if (!checkAddress(address, ss58Format)[0]) {
-    throw new SDKErrors.AddressInvalidError(address, name)
+  if (!checkAddress(input, ss58Format)[0]) {
+    throw new SDKErrors.AddressInvalidError(input)
   }
-  return true
 }
 
 /**
- * Validates the format of the given blake2b hash via regex.
+ * Type guard to check whether input is an SS58 address with our prefix of 38.
  *
- * @param hash Hash string to validate for correct Format.
- * @param name Contextual name of the address, e.g. "claim owner".
- * @returns Boolean whether the given hash string checks out against the Format.
+ * @param input Address string to validate for correct format.
+ * @returns True if input is a KiltAddress, false otherwise.
  */
-export function validateHash(hash: string, name: string): boolean {
-  if (typeof hash !== 'string') {
-    throw new SDKErrors.HashTypeError()
-  }
-  const blake2bPattern = new RegExp('(0x)[A-F0-9]{64}', 'i')
-  if (!hash.match(blake2bPattern)) {
-    throw new SDKErrors.HashMalformedError(hash, name)
+export function isKiltAddress(input: unknown): input is KiltAddress {
+  try {
+    verifyKiltAddress(input)
+    return true
+  } catch {
+    return false
   }
-  return true
 }
 
 /**
- * Validates the signature of the given signer address against the signed data.
+ * Validates the format of a hex string via regex.
  *
- * @param data The signed string of data.
- * @param signature The signature of the data to be validated.
- * @param signer Address of the signer identity.
- * @returns Boolean whether the signature is valid for the given data.
+ * @param input Hex string to validate for correct format.
+ * @param byteLength Expected length of hex in bytes. Defaults to 32.
+ * @param prefixed Whether the hex string is expected to be prefixed with '0x'. Defaults to true.
  */
-export function validateSignature(
-  data: string,
-  signature: string,
-  signer: KiltAddress
-): boolean {
-  if (
-    typeof data !== 'string' ||
-    typeof signature !== 'string' ||
-    typeof signer !== 'string'
-  ) {
-    throw new SDKErrors.SignatureMalformedError()
+export function verifyIsHex(
+  input: unknown,
+  byteLength = 32,
+  prefixed = true
+): void {
+  if (typeof input !== 'string') {
+    throw new SDKErrors.HashTypeError()
   }
-  if (!verify(data, signature, signer)) {
-    throw new SDKErrors.SignatureUnverifiableError()
+  const hexStringPattern = new RegExp(
+    `^${prefixed ? '0x' : ''}[A-F0-9]{${Math.round(byteLength) * 2}}$`,
```

Comment:
Went overkill with it again :D But I guess now you can verify any hex there is.
Fine for me

Reply:
as we discussed, we will use isHex from polkadot/utils.